scripts/updates/utility.sh

```diff
@@ -0,0 +1,105 @@
+termux_pkg_upgrade_version() {
+	if [ -z "$1" ]; then
+		echo "Usage: termux_pkg_upgrade_version <version to upgrade to (with epoch if required)>"
+		exit 1
+	fi
+
+	local latest_version="$1"
+
+	if [ "$BUILD_PACKAGES" = "false" ]; then
+		echo "Package '${TERMUX_PKG_NAME}' needs to be updated to $(echo "${latest_version}" | cut -d':' -f2)."
+	else
+		echo "Updating '${TERMUX_PKG_NAME}' to $(echo "${latest_version}" | cut -d':' -f2)."
+
+		sed -i \
+			"s/^\(TERMUX_PKG_VERSION=\)\(.*\)\$/\1${latest_version}/g" \
+			"${BASEDIR}/../../packages/${TERMUX_PKG_NAME}/build.sh"
+		sed -i \
+			"/TERMUX_PKG_REVISION=/d" \
+			"${BASEDIR}/../../packages/${TERMUX_PKG_NAME}/build.sh"
+
+		# Update checksum
+		if [ "${TERMUX_PKG_SHA256[*]}" != "SKIP_CHECKSUM" ] && [ "${TERMUX_PKG_SRCURL: -4}" != ".git" ]; then
+			echo n | "${BASEDIR}/../bin/update-checksum" "${TERMUX_PKG_NAME}" || {
+				echo "Warning: failed to update checksum for '${TERMUX_PKG_NAME}', skipping..."
+				git checkout -- "${BASEDIR}/../../packages/${TERMUX_PKG_NAME}"
+				git pull --rebase
+				return
+			}
+		fi
+
+		echo "Trying to build package '${TERMUX_PKG_NAME}'."
+		if "${BASEDIR}/../run-docker.sh" ./build-package.sh -a aarch64 -I "${TERMUX_PKG_NAME}" &&
+			"${BASEDIR}/../run-docker.sh" ./build-package.sh -a arm -I "${TERMUX_PKG_NAME}"; then
+			if [ "$GIT_COMMIT_PACKAGES" = "true" ]; then
+				git add "${BASEDIR}/../../packages/${TERMUX_PKG_NAME}"
+				git commit \
+					-m "${TERMUX_PKG_NAME}: update to $(echo "${latest_version}" | cut -d':' -f2)" \
+					-m "This commit has been automatically submitted by Github Actions."
+			fi
+
+			if [ "$GIT_PUSH_PACKAGES" = "true" ]; then
+				git pull --rebase
+				git push
+			fi
+
+		else
+
+			echo "Warning: failed to build '${TERMUX_PKG_NAME}'."
+			git checkout -- "${BASEDIR}/../../packages/${TERMUX_PKG_NAME}"
+		fi
+	fi
+}
+
+# Default algorithm to use when TERMUX_PKG_AUTO_UPDATE is set to true but no termux_pkg_auto_update
+# function is defined.
+termux_pkg_auto_update() {
```

Comment:
Rename function to `termux_github_auto_update` and move to `scripts/updates/termux_github_auto_update.sh`

This is because after this get's merged, I have plans to work on `termux_repology_auto_update`

Reply:
What does name `termux_github_auto_update` imply? Does this function should only auto update projects hosted on github?

Reply:
I feel it would make maintaining it easier if GitHub, Repology (to be added in future) and other update scripts should be kept at different places. The default `termux_pkg_auto_update` will just do the work of choosing between what update script to use depending on `$TERMUX_PKG_SRCURL`

Reply:
@thunder-coding if you do not mind, I would like to work on `termux_repology_auto_update`. If you are fine with it then please guide me what would you do for repology update...

Reply:
Repology does have an API too, did you check out its docs. Probably there's no reason to scrap the website.
https://repology.org/api

Reply:
That doesn't provide information per repository basis, so I decided to write my own scrapper.